lndboss/.bosgui/umbrel/credentials.json

```diff
@@ -0,0 +1,5 @@
+{
+  "cert_path": "/home/node/.lnd/tls.cert",
+  "macaroon_path": "/home/node/.lnd/data/chain/bitcoin/mainnet/admin.macaroon",
+  "socket": "10.21.21.9:10009"
```

Comment:
Is it possible to rely on configuration passed in via environment variables? Should Bitcoin's container IP change in the future (or become a dynamic IP), then lndboss will break.
This also goes for where you have `mainnet` hardcoded above. This will break in perhaps an unexpected way for users (like myself) who run a node using `regtest`.

Reply:
Hey Steve,

These values can be changed from the UI, the app supports running multiple nodes from a single UI, after logging in, you can click the menu on the top left and authenticate a new node. 

Without doing adding that credentials.json file, the app can read LND creds directly, the only problem is with the socket. It reads the LND_IP/hostname from lnd.conf file, however in the latest version of Umbrel, those settings were removed from the conf file, to get around it, I could do something like this in the docker-compose file:
```
extra-hosts:
 - localhost:${UMBREL_LND_IP}
 - umbrel.local:${UMBREL_LND_IP}
```
And it resolves the IP to localhost. 
This works on a custom Umbrel install on AWS, idk if it works on a raspberry Pi though. 


Let me know what you prefer.

Reply:
I see - you use this `credentials.json` file as a place to store the connection details... As we currently don't have a clean DNS setup for containers, I think using the extra hosts is the best idea atm.

Could you perhaps also write `credentials.json` on first boot of lndboss and write out state like the bitcoin network (ie check that the file doesn't exist and if not, use env. vars. and write them out). That way for users using eg regtest, when they install lndboss, 'it just works'.

@lukechilds Do you have any ideas here?

Reply:
Yeah, that file is used to point the app to the creds if it can't find itself or control a node remotely as it can also take base64 macaroon and cert. 

I think the extra hosts way works ok, if you could test it on a Pi. 

I could see if I can do it another way in a future update maybe, if that's ok? Since there is a workaround to update the creds path via the UI if it's necessary.

Reply:
Yeah these values need to be read from environment variables. It isn't safe to hardcode the IPs we currently use and just assume they won't change in the future.

Ideally LndBoss would accept env vars for LND connection details like Ride the Lightning does here:

https://github.com/getumbrel/umbrel-apps/blob/642a4bd52a2cfbb2431a84b1e9ea1b9a8b9d2273/ride-the-lightning/docker-compose.yml#L28-L31

Alternatively as a quick fix you could add an entrypoint to your container that reads these values from the env vars and writes them out to the config file, then starts LndBoss.